src/hooks/useStore_login.js

```diff
@@ -0,0 +1,24 @@
+import create from 'zustand';
+import {persist} from 'zustand/middleware';
+
+const loginStore = set => ({
+  login: {token: '', credentials: {username: '', password: ''}},
+
+  setToken: token => {
+    set(state => ({
+      login: {...state.login, token: token},
+    }));
+  },
+  setCredentials: (username, password) => {
+    set(state => ({
+      login: {
+        token: state.login.token,
+        credentials: {...state.login.credentials, username: username, password: password},
+      },
+    }));
+  },
+});
+
+const useStoreLogin = create(persist(loginStore, {name: 'login'}));
```

Comment:
Just a picky note: if you rename the useStore,  it's a common pattern zu name it something like `useLogin` or `useLoginStore`. 
Also, I would like to suggest that you name your file the same as your store. e.g. `userLogin.js` or `userLoginStore.js`

Reply:
Thank you for suggestion, I'll change it.